### ISSUE TITLE
Text scanning (OCR) not working after scan

### ISSUE THREAD

**Issue by dieTausendsassa** [holtwick/pdfify#67]
Please describe how the error can be reproduced:
1) Scan a document/page from a Scanner
2) Got to "Texterkennung"
3) Text scanning OCR don't work :-(

After saving and reopening the scanned document/page the OCR function works ...

App Version Info:
de.holtwick.mac.PDFify@3.3.2+130 WGAWN9BYE50VF9GW45BPVN60WG

**Comment by luettfite** [comment]
I can confirm this, same issue here.

**Comment by czux** [comment]
Is there an update on this? I found out about the problem after scanning roughtly 3000 pages only to find there's OCR text in the PDFs. The nasty thing is that after doing a scan OCR seems to be working okay but it's not. 
Log file reports "OCR done..."  

When opening a PDF directly Tesseract will do a proper OCR.

**Comment by holtwick** [comment]
This should be fixed in the next release. Please verify by using the beta version: https://pdfify.app/help#beta

**Comment by luettfite** [comment]
I tested with beta version 3.3.5-144. Unfortunately the issue is still the same. OCR doesn't work directly after scan. I have to save, close and open the document again. After this OCR works.
The problem seems to be related to the type of the scan. I have this problem, if I use the types "Text" or "Farbe" (Color) in the scanner dialog. If I use type "Schwarzweiß" (B&W), then OCR works on the fly, as expected.

**Comment by czux** [comment]
I can confirm all of this. I tested with version 3.3.5-142 and have experienced the exact same thing. Selecting "Schwarzweiß" (Greyscale) in the Scanner Dialog delivers an OCR as expected. Selecting "Farbe" (Color) or "Text" (1-bit B&W) leaves me with no OCR at all. I'm using an EPSON WorkForce Pro WF-C5710 DWF. @luettfite what Scanner do you use?

**Comment by czux** [comment]
When using EPSON Scan to scan the documents first (b/w, graysclae, color) and then feeding them to PDFify, OCR works as expected.

Only when doing the Scan from PDFify itself, it will fail for 2 of the 3 available color options.

**Comment by holtwick** [comment]
Thanks for the details, this helps a lot. I test on a Canon and was not able to reproduce the problem, even when using different modes. Black & white seems to have general issues and is not working for me at all. To me this looks like a macOS bug. However, could you please share your specific scanner options and OCR engine used? Thanks!

![20220311-111615-capture-holtwick@2x](https://user-images.githubusercontent.com/49557/157851005-8e1f9102-c8e6-4a7b-bae5-e97c6fa6d7be.png)

**Comment by luettfite** [comment]
I use Tesseract OCR German:
![1_Tessact_OCR_DE](https://user-images.githubusercontent.com/18640922/157936141-4363970c-8aa5-40b5-8489-030764fb00df.jpg)

My settings in the scanner dialog, where OCR after scan succeeds:
![3_Scan_Settings_Grey](https://user-images.githubusercontent.com/18640922/157936256-b6b5e296-5905-4164-8697-f5b3f09d121e.jpg)

My settings in the scanner dialog, where OCR after scan fails:
![2_Scan_Settings_Text](https://user-images.githubusercontent.com/18640922/157936246-e26bb748-ca71-421f-8c68-34d1955f1536.jpg) ![4_Scan_Settings_Color](https://user-images.githubusercontent.com/18640922/157936270-df4a97bf-af7b-4e5b-89ee-d6366ba860ab.jpg)

I also tested with other DPI values, but there was no difference in the OCR behavior.

Then I did another test and changed the engine to Apple Vision OCR German:
![5_AppleVision_OCR_DE](https://user-images.githubusercontent.com/18640922/157937153-37ec4da9-89b0-40cc-9092-f511a465d58e.jpg)

Now OCR after scan did work for all three types. So the OCR engine seems to make a difference too.

**Comment by czux** [comment]
For my setup it is exactly like @luettfite descibed it. I use the same settings and can confirm all of the observations.

**Comment by holtwick** [comment]
Thanks for the feedback. I'm still investigating, your comments will help.

**Comment by czux** [comment]
Using 3.3.5 (145) with Tesseract OCR I can still confirm all of the above. 

Assuming that the internal image passing from scan into PDFify is happening in TIFF format, I tried the following. Using only the macOS scan dialog outside PDFify, I produced a color and a grayscale PDF as well as a color and a grayscale TIFF. 

These are the results when dragging each of those into PDFify (only one at a time of course):

PDF color: OCR okay
PDF grayscale: OCR okay
TIFF color: **no OCR**
TIFF grayscale: OCR okay

In case of the color TIFF the OCR progress bar only flickers for a fraction of a second. In all other cases it takes a few seconds and shows proper OCR progress.

Maybe that is a useful piece of info to solve this issue.
